projects/evm-compatible-bridge/federator.md

```diff
@@ -0,0 +1,245 @@
+
+## Coordinator service
+
+The coordinator service will serve the function of the smart contract in the Hathor side.
+It will listen for transactions on the MultiSig wallet, keep a comprehensive persistent database so we do not process the same transaction twice and will gather signatures from the federation.
+It will be started with the MultiSig wallet config but it will not be a participant of the wallet.
+
+During the startup we need to check all existing transactions on the MultiSig address, if the transaction was already processed we can safely ignore it.
+After the startup we can start to listen for transactions using the queue plugin which enqueues events like new transactions made to one of our addresses.
+
+The coordinator service will provide a REST API for the federators and admins to interact with it.
+
+### Processing Hathor transactions
+
+When first starting the process it will iterate over all transactions in the MultiSig wallet and process them.
+After this, each new message will be enqueued and processed in order of arrival.
+Obs: The headless wallet offers a message queue extension to enqueue events e.g. new transactions.
+
+First we need to check the transaction is compliant with the request format if not, we can save it as processed and continue.
+If the transaction is already on tha database we can skip it, if not we need to check the federation contract to see if it was already processed.
+Obs: This read operation on the smart contract does not require a transaction to be sent, we can read the state from the EVM chain with 0 cost.
+
+The request format for a transaction is very simple:
+
+- The first output should be a data output, the data should be the destination address on the EVM.
+- There should only be 1 token being sent to the MultiSig wallet.
+
+To check this format we will call the headless api to check the transaction data and validate based on the response.
+
+Since this transaction was made from Hathor it will be saved as a Hathor request on the database with the following fields:
+
+- Hathor tx_id
+  - With the tx_id, so the federators can inspect the contents.
+- Hathor token uid
+  - The equivalent EVM token can be found on the smart contract storage (it will be created if this is the first time we see this token)
+- Amount of tokens being sent to the MultiSig wallet
+- destination address (on EVM chain)
+- Height
+  - Height of the first block that confirms the transaction.
+  - Transactions in Hathor do not have a height but we can use the `first_block_height` metadata.
+
+With this we can safely wait for 5 blocks on Hathor to have passed so we can create the proposal.
+There is an event `best-block-update` which tells the current height of the network.
+This proposal will be available for the polling API as soon as it is created.
+
+### API
```

Comment:
These APIs are on the coordinator, right?

Reply:
Correct

Reply:
Ok, please make it clear on the text.

Reply:
The header was changed to "Coordinator API"